packages/normalizr/src/normalize.ts

```diff
@@ -16,15 +16,18 @@ const visit = (
   addEntity: any,
   visitedEntities: any,
 ) => {
-  if (!value || !schema || !['function', 'object'].includes(typeof schema)) {
+  if (!value || !schema) {
     return value;
   }
 
+  // serializable
+  if (typeof schema === 'function' && !schema.normalize) {
+    return new schema(value);
+  }
```

Comment:
you may want to check for `schema.prototype` as well, unless the intention is for a non-constructible schema to throw

Reply:
what would be a valid check that something is nonconstructable? BigInt.prototype i'm getting {}

Reply:
The prototype check covers concise methods and arrow functions. There’s no better way to weed out things like BigInt and Symbol, though.

Reply:
Any insight into why BigInt is designed like that? lol. makes things hard.

Anyway, landed this for now since those aren't technical supported so they should throw errors; an improvement would be replacing those errors with a more helpful message, but that is not a regression so can be done for a future time.

Reply:
Really i'd like to just call the function if it doesn't want new, but it doesn't seem like there's a way to really make that work.

Reply:
tbh the most straightforward pattern is probably just "call a function", and they can do `() => new x()` if that's what they want.